roles/pid_generator/files/gen_pids.py

```diff
@@ -206,8 +206,8 @@ def openLocalPidDB():
 
 try:
     (db_con, db_cur) = openLocalPidDB()
-    outFile = open(urn_target_xml_name, "wb+")
-    rawFile = open(raw_target_name, "wb+")
+    outFile = open(urn_target_xml_name, "w+")
+    rawFile = open(raw_target_name, "w+")
```

Comment:
It's not great to do
```
whatnot_file = open("file.txt", "w+")
print("sisältöä", file=whatnot_file)
...
whatnot_file.close()
```
because of a number of reasons, e.g. file not getting properly closed if the program crashes and the risk of some contents not being written to disk even if the file is properly closed being higher.

Instead the modern and error-proof(er) way would be to do
```
with open(urn_target_xml_name, "w", encoding="utf-8") as out_file:
    out_file.write("sisältöä")
    ...
```
with the correct encodings etc of course. I'm not quite sure what those would be: the XML header seems to say ASCII, but can that really be true? I also assumed that we don't need the + in w+, as there didn't seem to be obvious use for it.

Not strictly necessary to do that here, as this is not illegal in Python 3, just riskier and slightly frowned upon.